File: WindowsFormsApplication1/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Security.Cryptography;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;

namespace WindowsFormsApplication1 {
    public partial class Form1 : Form {
        public Form1() {
            InitializeComponent();
        }

        private void Form1_Load(object sender, EventArgs e) {
        }

        private void button1_Click(object sender, EventArgs e) {
            MessageBox.Show(vncpass.EncryptPassword("xxxxxxx"));
            return;

            var a = "E3D7A8FC2C1FE8E677";
            var b = vncpass.DecryptPassword("E3D7A8FC2C1FE8E677");
            var c = vncpass.EncryptPassword(b);
            var d = vncpass.DecryptPassword(c);

            //MessageBox.Show(b);
            //MessageBox.Show(a + "\n" + c + "\n" + d);

            var m = vncpass.EncryptPassword("xxxx\0\0");
            var n = vncpass.EncryptPassword("xxxx");
            MessageBox.Show(m + "\n" + n);


            //MessageBox.Show(vncpass.EncryptPassword("794613"));
            //MessageBox.Show(vncpass.DecryptPassword(vncpass.EncryptPassword("794613")));
        }
    }

    class vncpass {
        public static string EncryptPassword(string plainPassword) {
            DES des = CreateDES();
            ICryptoTransform cryptoTransfrom = des.CreateEncryptor();

            plainPassword = plainPassword + "\x0000\x0000\x0000\x0000\x0000\x0000\x0000\x0000";
            plainPassword = plainPassword.Length > 8 ? plainPassword.Substring(0, 8) : plainPassword;
            byte[] data = Encoding.ASCII.GetBytes(plainPassword);
            byte[] encryptedBytes = cryptoTransfrom.TransformFinalBlock(data, 0, data.Length);

            return ByteArrayToHex(encryptedBytes) + "00";
        }

        public static string DecryptPassword(string encryptedPassword) {
            DES des = CreateDES();
            ICryptoTransform cryptoTransfrom = des.CreateDecryptor();
            byte[] data = HexToByteArray(encryptedPassword.Substring(0, encryptedPassword.Length - 2));
            byte[] decryptedBytes = cryptoTransfrom.TransformFinalBlock(data, 0, data.Length);

            return Encoding.ASCII.GetString(decryptedBytes);
        }

        private static DES CreateDES() {
            byte[] key = { 0xE8, 0x4A, 0xD6, 0x60, 0xC4, 0x72, 0x1A, 0xE0 };
            DES des = DES.Create();
            des.Key = key;
            des.IV = key;
            des.Mode = CipherMode.ECB;
            des.Padding = PaddingMode.Zeros;
            return des;
        }

        private static string ByteArrayToHex(byte[] bytes) {
            char[] c = new char[bytes.Length * 2];
            byte b;
            for (int i = 0;i < bytes.Length;i++) {
                b = (byte)(bytes[i] >> 4);
                c[i * 2] = (char)(b > 9 ? b + 0x37 : b + 0x30);
                b = (byte)(bytes[i] & 0xF);
                c[(i * 2) + 1] = (char)(b > 9 ? b + 0x37 : b + 0x30);
            }
            return new string(c);
        }

        private static byte[] HexToByteArray(string hex) {
            if (hex.Length % 2 == 1) {
                throw new Exception("The binary key cannot have an odd number of digits");
            }
            byte[] arr = new byte[hex.Length >> 1];
            for (int i = 0;i < (hex.Length >> 1);++i) {
                arr[i] = (byte)(((hex[i << 1] - (hex[i << 1] < 58 ? 48 : 55)) << 4) + (hex[(i << 1) + 1] - (hex[(i << 1) + 1] < 58 ? 48 : 55)));
            }
            return arr;
        }
    }

}

File: WindowsApplication1/Form1.vb
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Linq
Imports System.Security.Cryptography
Imports System.Text
Imports System.Threading.Tasks
Imports System.Windows.Forms

Namespace WindowsFormsApplication1
    Partial Public Class Form1
        Inherits Form

        Private Sub button1_Click(sender As Object, e As EventArgs)
            Dim a = "E3D7A8FC2C1FE8E677"
            Dim b = vncpass.DecryptPassword("E3D7A8FC2C1FE8E677")
            Dim c = vncpass.EncryptPassword(b)
            Dim d = vncpass.DecryptPassword(c)

            'MessageBox.Show(b);
            'MessageBox.Show(a + "\n" + c + "\n" + d);

            Dim m = vncpass.EncryptPassword("xxxx" & vbNullChar & vbNullChar)
            Dim n = vncpass.EncryptPassword("xxxx")
            MessageBox.Show(Convert.ToString(m & Convert.ToString(vbLf)) & n)


            'MessageBox.Show(vncpass.EncryptPassword("794613"));
            'MessageBox.Show(vncpass.DecryptPassword(vncpass.EncryptPassword("794613")));
        End Sub
    End Class

    Class vncpass
        Public Shared Function EncryptPassword(plainPassword As String) As String
            Dim des As DES = CreateDES()
            Dim cryptoTransfrom As ICryptoTransform = des.CreateEncryptor()

            plainPassword = plainPassword & Convert.ToString(vbNullChar & vbNullChar & vbNullChar & vbNullChar & vbNullChar & vbNullChar & vbNullChar & vbNullChar)
            plainPassword = If(plainPassword.Length > 8, plainPassword.Substring(0, 8), plainPassword)
            Dim data As Byte() = Encoding.ASCII.GetBytes(plainPassword)
            Dim encryptedBytes As Byte() = cryptoTransfrom.TransformFinalBlock(data, 0, data.Length)

            Return ByteArrayToHex(encryptedBytes) & Convert.ToString("00")
        End Function

        Public Shared Function DecryptPassword(encryptedPassword As String) As String
            Dim des As DES = CreateDES()
            Dim cryptoTransfrom As ICryptoTransform = des.CreateDecryptor()
            Dim data As Byte() = HexToByteArray(encryptedPassword.Substring(0, encryptedPassword.Length - 2))
            Dim decryptedBytes As Byte() = cryptoTransfrom.TransformFinalBlock(data, 0, data.Length)

            Return Encoding.ASCII.GetString(decryptedBytes)
        End Function

        Private Shared Function CreateDES() As DES
            Dim key As Byte() = {&HE8, &H4A, &HD6, &H60, &HC4, &H72, _
                &H1A, &HE0}
            Dim des__1 As DES = DES.Create()
            des__1.Key = key
            des__1.IV = key
            des__1.Mode = CipherMode.ECB
            des__1.Padding = PaddingMode.Zeros
            Return des__1
        End Function

        Private Shared Function ByteArrayToHex(bytes As Byte()) As String
            Dim c As Char() = New Char(bytes.Length * 2 - 1) {}
            Dim b As Byte
            For i As Integer = 0 To bytes.Length - 1
                b = CByte(bytes(i) >> 4)
                c(i * 2) = CChar(ChrW(If(b > 9, b + &H37, b + &H30)))
                b = CByte(bytes(i) And &HF)
                c((i * 2) + 1) = CChar(ChrW(If(b > 9, b + &H37, b + &H30)))
            Next
            Return New String(c)
        End Function

        Private Shared Function HexToByteArray(hex As String) As Byte()
            If hex.Length Mod 2 = 1 Then
                Throw New Exception("The binary key cannot have an odd number of digits")
            End If
            Dim arr As Byte() = New Byte((hex.Length >> 1) - 1) {}
            For i As Integer = 0 To (hex.Length >> 1) - 1
                arr(i) = CByte(((hex(i << 1) - (If(hex(i << 1) < 58, 48, 55))) << 4) + (hex((i << 1) + 1) - (If(hex((i << 1) + 1) < 58, 48, 55))))
            Next
            Return arr
        End Function
    End Class

End Namespace

